### ISSUE TITLE
Example doesn't use inputted arguments

### ISSUE THREAD

**Issue by ewellinger** [dmatrix/jsd-mlflow-examples#7]
In the `keras_nn_model.py` example it doesn't appear as though you are actually using any of the passed parameters.  For example on line 68 you are setting the `drop_rate` to 0.5.  No matter what you set the drop_rate to through the command line it will always use that hardcoded value of 0.5.

```python
drop_rate = 0.5
...
parser.add_argument("--drop_rate", help="Drop rate", nargs='?', action='store', default=0.5, type=float)
...
print("drop_rate", args.drop_rate)
...
model = build_model(in_dim=input_dim, drate=drop_rate, out=output)  # This is going to be the 0.5 value set on line 68, not the value printed out on line 76
...
mlflow.log_param("drop_rate", args.drop_rate)  # This is logging the value that wasn't used
```

**Comment by ewellinger** [comment]
Also looks like there are some import issues in this example.  You are called `mlflow.log_param` but `mlflow` was never imported in this script.

I noticed that you are logging the time to run on line 109 outside of the `start_run()` context block.  Should that be moved inside the run context?

**Comment by dmatrix** [comment]
> In the `keras_nn_model.py` example it doesn't appear as though you are actually using any of the passed parameters. For example on line 68 you are setting the `drop_rate` to 0.5. No matter what you set the drop_rate to through the command line it will always use that hardcoded value of 0.5.
> 
> ```python
> drop_rate = 0.5
> ...
> parser.add_argument("--drop_rate", help="Drop rate", nargs='?', action='store', default=0.5, type=float)
> ...
> print("drop_rate", args.drop_rate)
> ...
> model = build_model(in_dim=input_dim, drate=drop_rate, out=output)  # This is going to be the 0.5 value set on line 68, not the value printed out on line 76
> ...
> mlflow.log_param("drop_rate", args.drop_rate)  # This is logging the value that wasn't used
> ```

Good point. Easily fixed by changing the code around...

args = parser.parse_args()
    
    drop_rate = args.drop_rate
    input_dim = args.input_dim
    bs = args.bs
    output = args.output
    epochs = args.epochs
    batch_size = args.train_batch_size

    print("drop_rate", args.drop_rate)
    print("input_dim", args.input_dim)
    print("size", args.bs)
    print("output", args.output)
    print("train_batch_size", args.train_batch_size)
    print("epochs", args.epochs)

It will take the defaults if not specified and value ones specified. Will update the code shortly, test it and close the issue. 

Thanks!

**Comment by dmatrix** [comment]
> Also looks like there are some import issues in this example. You are called `mlflow.log_param` but `mlflow` was never imported in this script.
> 
> I noticed that you are logging the time to run on line 109 outside of the `start_run()` context block. Should that be moved inside the run context?

Actually, it's fine. We want to measure the time it took to build, train, & log.

**Comment by dmatrix** [comment]
Here is run with the change:

C02RD34PG8WM:jsd-mlflow-examples jules$ python keras/keras_nn_model.py --output=128 --epochs=10 --drop_rate 0.6
Using TensorFlow backend.
drop_rate 0.6
input_dim 20
size 1000
output 128
train_batch_size 128
epochs 10
2018-11-05 13:01:39.285176: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
12/12 [==============================] - 0s 5ms/step
Test loss: 0.6578670144081116
Test accuracy: 0.6666666865348816
Predictions for Y:
[[0.55530953]
 [0.54123324]
 [0.5369644 ]
 [0.5292471 ]
 [0.55531996]]
_________________________________________________________________
Layer (type)                 Output Shape              Param #
=================================================================
dense_1 (Dense)              (None, 128)               2688
_________________________________________________________________
dropout_1 (Dropout)          (None, 128)               0
_________________________________________________________________
dense_2 (Dense)              (None, 128)               16512
_________________________________________________________________
dropout_2 (Dropout)          (None, 128)               0
_________________________________________________________________
dense_3 (Dense)              (None, 1)                 129
=================================================================
Total params: 19,329
Trainable params: 19,329
Non-trainable params: 0
_________________________________________________________________
This model took 1.3896019458770752 seconds to train and test.
C02RD34PG8WM:jsd-mlflow-examples jules$

**Comment by ewellinger** [comment]
Sweet!  Looking forward to incorporating mlflow into some of our projects.

**Comment by dmatrix** [comment]
> Sweet! Looking forward to incorporating mlflow into some of our projects.

Excellent. Look at some the blogs that show good examples of MLflow usage.

- https://databricks.com/blog/2018/07/10/how-to-use-mlflow-tensorflow-and-keras-with-pycharm.html
- https://databricks.com/blog/2018/08/23/how-to-use-mlflow-to-experiment-a-keras-network-model-binary-classification-for-movie-reviews.html

- https://databricks.com/blog/2018/09/21/how-to-use-mlflow-to-reproduce-results-and-retrain-saved-keras-ml-models.html

- https://databricks.com/blog/2018/08/16/introducing-mlflow-apps-a-repository-of-sample-applications-for-mlflow.html
